rfcs/0130-stalled-rfcs.md

```diff
@@ -0,0 +1,145 @@
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+## More Research Required
+
+If an RFC is opened and discussion reveals that more research is required, the author may now return their RFC to draft status to indicate that they are working on an update.
+
+This is already frequently done today, but this RFC codifies this process. The alternative, which is sometimes seen is that the NixOS RFC Steering Committee keeps pinging the RFC for months trying to move the RFC forward which is annoying and unproductive.
+
+## Can't Find Shepherds
+
+If an RFC is opened but a team of shepherds can't be assembled the RFC will now be closed.
+
+Currently there is no process here and the NixOS RFC Steering Committee is unsure what to do and tends to annoyingly ping these RFCs with no positive outcome for a few months then improvise. This RFC proposes a well-defined process to ensure that everyone understands the process.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Closing a PR may be demotivating.
+
+Struggling to find shepherds for an RFC may already be demotivating and having the RFC closed can be an additional flag that the RFC is struggling. This may be demoralizing to the RFC author.
+
+This issue is mitigated by having clear and well-thought-out messages to make it clear what is occurring at this stage of the process. This is hopefully better overall than the previously poorly defined state of forever waiting for more shepherds to appear.
+
+## Closing a PR may reduce visibility for the RFC.
+
+Leaving PRs open may increase the chance that a potential shepherd finds the RFC. Closing RFCs that can't find shepherds may worsen the thing that ails them.
+
+It is the author's opinion that having dozens of open RFCs will also reduce visibility from the ones that do have more interest and are more likely to find a full shepherd team. Therefore this is considered an acceptable cost. The addition of the `status: insufficient interest` label may also help to bring attention to these RFCs that could benefit from would-be shepherds.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+
+Doing nothing leaves the behaviour to be taken when an RFC can not find a shepherd team undefined. This leads to inconsistent handling by the NixOS RFC Steering Committee and leaves community members unable to understand what to expect.
+
+## Leave the PR open and add a label
+
+This can reduce the burden from the NixOS RFC Steering Committee however it still clutters the list of "Open" RFCs with these RFCs that are not seeing forward progress. Ultimately it is a decision of definition of what "Open" means and this RFC takes the stance that if an RFC is stalled for too long it makes sense to remove it from the default search for "Open" RFCs.
```

Comment:
Argument in favour of this approach: we have had _two_ separate discussions about stale PR handling, with the end result being «label but don't close». I think consistency is good.

Note that we have few enough RFCs that a round of RFC SC pings ensures that probably-live ones are indeed on top in the list sorted by last-updated.

Reply:
The main tangible difference I see is if we want these to appear in the default search which on GitHub is "open order by created descending". The current proposal will hide these whereas this alternative will interleave these with other open and active PRs.

RFC SC can easily pin the right search to see what they want so it doesn't really matter to them.

Personally I have a small preference for hiding them to focus attention on "active" PRs but I definitely see the argument for trying to get more attention to these PRs that need attention most, even if that thins out the attention for other PRs a bit.

But I don't have a strong opinion either way. I avoided responding to this comment earlier in hopes that others would add their opinion but it seems like there isn't that much debate here.

Reply:
We have RFCs#51 where various «clean list» argument were presentde but ended up rejected, and RFCs#124 seems to show that sometimes there is a lot of support to make policy similar to RFCs#51. It _is_ true that RFC proposals are slighlty different, but then I guess we need a detailed argument why it is better to pick a different policy than for Nixpkgs and Nix repos here, and I guess a dedicated large announcement. Otherwise it would be natural to expect that things get labeled not closed.

I would expect people look for «what is active» and sort by activity, or «what discussions have I missed» (sorting by creation time) and then it is not always clear if RFCs lacking shepherds are out of scope.

It does look like right now «draft» label seems to mix waiting-on-author and lacking-shepherds right now… making this available at glance will surely be useful.

For closing clearly-not-going-anywhere… maybe add a transition where force closing can be done if some _person_ bothers to ask the author what's the plan and does not get any response at all in a month? I guess switch of authorship requires resubmission in any case.

Reply:
I'm usually against "clean list" arguments. But that is more because for issues it is helpful to find the existing ones. For RFCs I'm not as sure, but maybe I'm over-valuing the "browsing" use case.

I think if we do the label approach then we end up where the author may give up after a while and close. Or we may have lots of open PRs where the author gave up and forgot.

Do you feel strongly?

Reply:
> for issues

I would say RFCs are also proposed as a reaction to a perceived issue with something we are doing!

> maybe I'm over-valuing the "browsing" use case.

I am making an even weaker objection: it is unclear how this use case is split among a few plausible variants (which are better served by different decisions).

> I think if we do the label approach then we end up where the author may give up after a while and close. Or we may have lots of open PRs where the author gave up and forgot.

Author withdrawing is fine. As I said, human-judgement-driven (without recommended timeline) check if author went missing is also fine and maybe it could be added as a text-only clarification (if authors are nowhere to be found, I guess it counts as the authors withdrawing themselves from the process, so it's just a footnote for an existing transition…).

> Do you feel strongly?

I feel strongly that fixed-delay-based closing of discussions within NixOS namespace at GitHub needs a very public announcement of intent and justification which explicitly shows that arguments from #51 are not applicable here (and then, well, lack of strong pushback in replies to that announcement).

But my position is more process-based: I have nothing against this decision if the heads-up announcement is there, stresses the difference, and is well received. (I guess I would prefer uniformity, but that's a weak preference and people mostly agreeing this case is separate would override it)

Reply:
I'd be happy to get more eyes on this but am not sure of the best avenue to get more eyes and opinions than this RFC What would you recommend?

Reply:
Maybe a Discource heads-up announcement that one part of the current RFC might be surprising for some? If you think it is a good idea, we could mention-ping the other shepherds to check for reasonability, maybe in the non-line-tied part of the discussion.

Reply:
Attempted to solicit more feedback here: https://discourse.nixos.org/t/stalled-rfcs-proposing-rfcs-with-no-interest/21952

Reply:
(reposted from where I previously put it, which was the wrong place)
I think an important distinction between auto-closing issues and this where the responsibility for getting it out of that state lies; with automatically closed issues, only people with write access to the repo can reopen them. In the case of RFCs, any group of people with enough interest can pick the RFC up and get it moving again, and the RFC Steering Committee will see and apply the administrivial changes.

Reply:
@7c6f434c and others. I have proposed a note to codify this check in the RFC document. Does this sound good to you?

https://github.com/NixOS/rfcs/pull/130/files#r990635602